proto/data.go

```diff
@@ -316,6 +317,69 @@ func (v *Value) SetProto(msg gogoproto.Message) error {
 	return nil
 }
 
+// SetTime encodes the specified time value into the bytes field of the
+// receiver and sets the tag.
+func (v *Value) SetTime(t time.Time) error {
+	var err error
+	if v.Bytes, err = t.MarshalBinary(); err != nil {
```

Comment:
FYI, this PR didn't change the encoding of timestamps. @vivekmenezes You're going to be touching some of the timestamp and timezone stuff again. Keep this code in mind when you do.